### PR TITLE
Add missing cases for stats refresh.

### LINKED ISSUE #11700: opt: add missing cases to list of when statistics are refreshed
In [this section](https://www.cockroachlabs.com/docs/stable/cost-based-optimizer.html#controlling-statistics-refresh-rate), we list the cases when statistics are refreshed:
1. When there are no statistics.
2. When it's been a long time since the last refresh, where "long time" is defined according to a moving average of the time across the last several refreshes.
3. After each mutation operation (INSERT, UPDATE, or DELETE), the probability of a refresh is calculated using a formula that takes the cluster settings shown below as inputs. These settings define the target number of rows in a table that should be stale before statistics on that table are refreshed. Increasing either setting will reduce the frequency of refreshes. In particular, `min_stale_rows` impacts the frequency of refreshes for small tables, while `fraction_stale_rows` has more of an impact on larger tables.

These three items are accurate, but the list is missing a couple of cases:
- After a successful `IMPORT` or `RESTORE` into the table.
- After any schema change affecting the table.

These cases have existed since automatic stats were first introduced, so all previous versions of the docs should be updated.

cc @ianjevans @stbof

### PR DESCRIPTION
**Pull request by stbof**
Fixes #11700. Issue addressed in lines 43-47. 
Remaining updates are for style conformance.

### CODE REVIEW

File: v21.2/cost-based-optimizer.md [hunk 1]
```diff
@@ -64,29 +67,26 @@ If you need to turn off automatic statistics collection, follow the steps below:
     > SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
     ~~~
 
-2. Use the [`SHOW STATISTICS`](show-statistics.html) statement to view automatically generated statistics.
+1. Use the [`SHOW STATISTICS`](show-statistics.html) statement to view automatically generated statistics.
 
-3. Delete the automatically generated statistics using the following statement:
+1. Delete the automatically generated statistics using the following statement:
 
     {% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.2/cost-based-optimizer.md:74]:
```suggestion
    {% include_cached copy-clipboard.html %}
```

File: v21.2/cost-based-optimizer.md [hunk 2]
```diff
@@ -64,29 +67,26 @@ If you need to turn off automatic statistics collection, follow the steps below:
     > SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
```
Comment (ianjevans) [inline v21.2/cost-based-optimizer.md:67]: To speed up build times, we've been replacing all `{% include copy-clipboard.html %}` tags with `{% include_cached copy-clipboard.html %}`. So I'm just noting them in this review.

**Review by ianjevans — APPROVED** [review]
We'll need to apply these same changes to the v21.2 and v20.2 versions of `cost-based-optimizer.md`. I also noted a few places where we should change the `include` to `include_cached`. Otherwise, looks good.

**Review by rytaft — APPROVED** [review]
:lgtm: thanks!

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11707#-:-Mk3Z8kn6ESZKYg8ESox:brfqlux)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)

___
*[v21.2/cost-based-optimizer.md, line 89 at r2](https://reviewable.io/reviews/cockroachdb/docs/11707#-Mk3YhAqEUeDmE4VAlpo:-Mk3YhAqEUeDmE4VAlpp:bhjubdu) ([raw file](https://github.com/cockroachdb/docs/blob/497b4e6aa85528c3c1b4f74e66a65cc7da0e58c2/v21.2/cost-based-optimizer.md#L89)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> CockroachDB does not support multi-column histograms yet. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/49698).
> ```

Did you intend to remove the link to the GitHub issue?

**Comment by stbof** [comment]
> :lgtm: thanks!
> 
> > Reviewed 1 of 1 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11707#-:-Mk3Z8kn6ESZKYg8ESox:brfqlux)_ status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)
> 
> _[v21.2/cost-based-optimizer.md, line 89 at r2](https://reviewable.io/reviews/cockroachdb/docs/11707#-Mk3YhAqEUeDmE4VAlpo:-Mk3YhAqEUeDmE4VAlpp:bhjubdu) ([raw file](https://github.com/cockroachdb/docs/blob/497b4e6aa85528c3c1b4f74e66a65cc7da0e58c2/v21.2/cost-based-optimizer.md#L89)):_
> 
> > ```gfm
> > {{site.data.alerts.callout_info}}
> > CockroachDB does not support multi-column histograms yet. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/49698).
> > ```
> 
> Did you intend to remove the link to the GitHub issue?

Yes. You will notice that I also removed the word "yet". I discussed this with @ianjevans.

In general documentation is a legal contract that describes the state of the product as it is, not as it may be in the future. That issue is already over a year old. In particular as the issue gets older, it does not reflect well on the company to have links to things we might have but haven't committed to. I will be bringing this up in the docs team meeting so that we clearly articulate a policy around this area.

**Pull request merged**